src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -139,10 +149,40 @@ public static StashDiffReport extractDiffs(String jsonBody) throws StashReportEx
                           
                           StashDiff diff = new StashDiff(type, path, source, destination);
                           
+                          // Add comment attached to the current line
                           JSONArray jsonCommentIds = (JSONArray) jsonLine.get("commentIds");
                           if (jsonCommentIds != null) {
+                            
                             for (Object objCommentId: jsonCommentIds.toArray()) {
-                              diff.addComment((long) objCommentId);
+                              long commentId = (long) objCommentId; 
+                              
+                              JSONArray jsonLineComments = (JSONArray) jsonDiff.get("lineComments");
+                              if (jsonLineComments != null) {
+                                for (Object objLineComment : jsonLineComments.toArray()) {
+                                  JSONObject jsonLineComment = (JSONObject) objLineComment;
+                                  
+                                  long lineCommentId = (long) jsonLineComment.get("id");
+                                  if (lineCommentId == commentId) { 
+                                  
+                                    String lineCommentMessage = (String) jsonLineComment.get("text");
+                                    long lineCommentVersion = (long) jsonLineComment.get("version");
+                                  
+                                    JSONObject objAuthor = (JSONObject) jsonLineComment.get("author");
+                                    if (objAuthor != null) {
+                                      
+                                      long authorId = (long) objAuthor.get("id");

```

Comment:
doesn't it make sense to use extractUser()?
https://developer.atlassian.com/static/rest/stash/3.11.3/stash-rest.html#idp1467264
emailAddress insteadof email, no?